### PR TITLE
Add "is_sandbox" to ReceiptInfo

### PR DESCRIPTION
**Pull request by qchenqizhi**
Let users know if it is a sandbox request through callback

### CODE REVIEW

**Comment by Sam-Spencer** [comment]
There are a number of ways available, outside of StoreKit, for an app to determine whether it is currently being tested or has been distributed outside of the App Store. I'm sorry, but I'm not sure I see any value in this beyond what is already available from the system.

For the time being, I am going to close this pull request. If you could elaborate on your motivations or use cases for this specific feature I will be more than happy to re-open the PR. Thank you!

**Comment by qchenqizhi** [comment]
Yes, there are many ways (for example: [SwifterSwift.UIApplicationExtensions](https://github.com/SwifterSwift/SwifterSwift/blob/master/Sources/SwifterSwift/UIKit/UIApplicationExtensions.swift)) to determine is in sandbox except one: App in reviewing, all methods will tell you the current is release and in App Store, but the actual sandbox is used.

File: SwiftyStoreKit/AppleReceiptValidator.swift [hunk 1]
```diff
@@ -110,9 +110,15 @@ public struct AppleReceiptValidator: ReceiptValidator {
 					sandboxValidator.validate(receiptData: receiptData, completion: completion)
 				} else {
 					if receiptStatus.isValid {
-						completion(.success(receipt: receiptInfo))
+                        if case .sandbox = self.service {
+                            var mutableReceiptInfo = receiptInfo
+                            mutableReceiptInfo["is_sandbox"] = "1" as AnyObject
```
Comment (Sam-Spencer) [inline SwiftyStoreKit/AppleReceiptValidator.swift]: I am hesitant to start adding additional information to the receipt, especially as some apps may not expect this data and it adds a layer of complexity for developers.

Instead, a better approach might be to make the `service` property public with a `private(set)`.

Even so, the comment above this clearly indicates you can check the provided receipt's status code for `21007`, to see if the app is being tested in the sandbox (esp. during app review).

**Comment by Sam-Spencer** [comment]
Does this relate to #525?

**Comment by qchenqizhi** [comment]
> Does this relate to #525?

I don't think so.

File: SwiftyStoreKit/AppleReceiptValidator.swift (hunk 1 above)
Comment (qchenqizhi) [reply SwiftyStoreKit/AppleReceiptValidator.swift]: It is mainly used for statistics. I think only make `service` get public will not work, should use the same `AppleReceiptValidator` instance and change service to `sandbox` and `validate()` again, or add current used `AppleReceiptValidator` to the completion closure

**Comment by qchenqizhi** [comment]
Added a new PR for this: [#536](https://github.com/bizz84/SwiftyStoreKit/pull/536)

**Pull request closed without merging**